lib/services/remote_sync_service.dart

```diff
@@ -223,10 +224,87 @@ class RemoteSyncService {
   }
 
   Future<void> _syncDeviceCollectionFilesForUpload() async {
-    final devicePathCollections =
-        await FilesDB.instance.getDevicePathCollections();
+    final FilesDB fileDb = FilesDB.instance;
+    final devicePathCollections = await fileDb.getDevicePathCollections();
     devicePathCollections.removeWhere((element) => !element.sync);
     await _createCollectionsForDevicePath(devicePathCollections);
+    final Map<String, Set<String>> unSyncedPathIdToLocalIDs =
+        await fileDb.getDevicePathIDToLocalIDMap(syncStatus: false);
+    /*
+       A) Check if mapping for localID already exist in the collection
+       B) Check if
+     */
+    for (final eachDevicePath in devicePathCollections) {
+      debugPrint("Processing ${eachDevicePath.name}");
+      final String pathID = eachDevicePath.id;
+      final Set<String> unSyncedLocalIDs =
+          unSyncedPathIdToLocalIDs[eachDevicePath.id] ?? {};
+      if (unSyncedLocalIDs.isNotEmpty && eachDevicePath.collectionID != -1) {
+        await fileDb.setCollectionIDForUnMappedLocalFiles(
+          eachDevicePath.collectionID,
+          unSyncedLocalIDs,
+        );
+
+        // mark IDs as already synced if corresponding entry is present in
+        // the collection. This can happen when a user has marked a folder
+        // for sync, then un-synced it and again tries to mark if for sync.
+        final Set<String> existingMapping = await fileDb
+            .getLocalFileIDsForCollection(eachDevicePath.collectionID);
+        final Set<String> commonElements =
+            unSyncedLocalIDs.intersection(existingMapping);
+        if (commonElements.isNotEmpty) {
+          debugPrint(
+            "${commonElements.length} files already existing in "
+            "collection ${eachDevicePath.collectionID} for ${eachDevicePath.name}",
+          );
+          await fileDb.insertPathIDToLocalIDMapping(
+            {pathID: commonElements},
+            conflictAlgorithm: ConflictAlgorithm.replace,
+            syncStatus: true,
+          );
+          unSyncedLocalIDs.removeAll(commonElements);
+        }
+
+        // At this point, the remaining unSyncedLocalIDs will need to create
+        // new file entries, where we can store mapping for localID and
+        // corresponding collection ID
+        if (unSyncedLocalIDs.isNotEmpty) {
+          debugPrint(
+            'Adding new entries for ${unSyncedLocalIDs.length} files'
+            ' for ${eachDevicePath.name}',
+          );
+          final filesWithPotentialCollectionID =
```

Comment:
add server log to capture potential inconsistent state.

Reply:
Simply `filesWithCollectionID` here?